src/Symfony/Component/Console/CommandLoader/ContainerCommandLoader.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Symfony\Component\Console\CommandLoader;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * Loads commands from a PSR-11 container.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class ContainerCommandLoader implements CommandLoaderInterface
+{
+    private $container;
+    private $commandNames;
+
+    public function __construct(ContainerInterface $container, array $commandNames)
+    {
+        $this->container = $container;
+        $this->commandNames = $commandNames;
```

Comment:
what's the point of having an array of command names when you could just have all these in the container ? Just for the `all` ?